presto-main/src/test/java/io/prestosql/tests/LogTestDurationListener.java

```diff
@@ -34,19 +36,23 @@
     private static final Duration SINGLE_TEST_LOGGING_THRESHOLD = Duration.valueOf("30s");
     private static final Duration CLASS_LOGGING_THRESHOLD = Duration.valueOf("1m");
 
-    private final Map<String, Long> started = new ConcurrentHashMap<>();
+    private final Map<String, TestData> started = new ConcurrentHashMap<>();
```

Comment:
IMO This is unnecessarily complex. I think we could just measure time from first start to last end (or to any end, we can log more than once; hey this is just tests).

Reply:
Can we just get rid of the TestNG Factory thing so we don't run into this problem at all?

Reply:
We should. Especially that they are indistinguishable in tests results (at least not when running tests in my IDE)

Reply:
I changed the regex test to not use a factory.  I added checks in the listener to throw, so no one can add a test like that again.

Without the change to the regex test you get this error:
```
java.lang.IllegalStateException: There already is a start record for test: io.prestosql.operator.scalar.TestRegexpFunctions
	at com.google.common.base.Preconditions.checkState(Preconditions.java:588)
	at io.prestosql.tests.LogTestDurationListener.beginTest(LogTestDurationListener.java:168)
	at io.prestosql.tests.LogTestDurationListener.onBeforeClass(LogTestDurationListener.java:135)
	at org.testng.internal.TestMethodWorker.invokeBeforeClassMethods(TestMethodWorker.java:167)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
        ...
```